### ISSUE TITLE
A queston about filelist

### ISSUE THREAD

**Issue by KongMingxi** [biomedia-mira/masf#10]
For PACS dataset, the txt files of each domain include three files. Such as cartoon, there are cartoon_crossval_kfold.txt, cartoon_train_kfold.txt and cartoon_test_kfold.txt.

I download the filelist from http://www.eecs.qmul.ac.uk/~dl307/project_iccv2017, as given in this code.

Which txt files should be put in '/path/to/image/filelist'?

**Comment by carrenD** [comment]
Each .txt file keeps a single file.

**Comment by hosseinshn** [comment]
I have a similar issue. Based on your code, it requires e.g. "cartoon.txt" but such a file does not exist in the provided link and as @KongMingxi said, there are multiple files for each source domain. Which file(s) did you use to produce the results?

**Comment by vihari** [comment]
I have a similar question. My baseline numbers, obtained from running your code for 10,000 steps with 1e-5 lr are much smaller than what is reported for DeepAll in the paper. Wonder if you have used also the val splits for training, please clarify. Also, is there any standardization on what augmentations to use? For example, you use left-right flip augment but https://github.com/HaohanWang/HEX_experiments/blob/master/PACS/datagenerator.py does not use any, does that make the comparisons unfair?

**Comment by vihari** [comment]
Do you use the training data such as the one used by MLDG codestack? https://github.com/yogeshbalaji/Meta-Learning-Domain-Generalization/tree/master/data/MLDG 
Which is making use of both train and val splits for training.

**Comment by carrenD** [comment]
Sorry for late reply, have you solved your problem? I didn't use val split for training. The meta-train and meta-test are from train split.

**Comment by sshan-zhao** [comment]
> Sorry for late reply, have you solved your problem? I didn't use val split for training. The meta-train and meta-test are from train split.

Hi, how do you select the model for evaluation on the target domain? It seems that you directly evaluate the model on the target domain every epoch, and then report the best performance.

**Comment by sshan-zhao** [comment]
> Sorry for late reply, have you solved your problem? I didn't use val split for training. The meta-train and meta-test are from train split.

And similarly, how do you select the best model on VLCS dataset?
Thanks.

**Comment by wildphoton** [comment]
According to the code, it directly uses the target domain data for validation and there is no train/valid split. It means that the best model is selected on the testing data from which the reported performance comes. Have I misunderstood anything?

**Comment by lukasuz** [comment]
> According to the code, it directly uses the target domain data for validation and there is no train/valid split. It means that the best model is selected on the testing data from which the reported performance comes. Have I misunderstood anything?

Any updates on this? I am also confused about the strategy of choosing the best model for final testing.

**Comment by carrenD** [comment]
Hi, yep, the best model are marked for training and analysis purpose, the report model selection is based on training time. Considering the dataset is small, separating a validation subset may still not representative. In DG literature, leave-one-domain-out also have did not split the validation set. Thanks!
